pennylane/qchem/factorization.py

```diff
@@ -14,31 +14,106 @@
 """
 This module contains the functions needed for two-electron tensor factorization.
 """
+from functools import partial
+
 import numpy as np
+import scipy as sp
 
 import pennylane as qml
 
-
-def factorize(two_electron, tol_factor=1.0e-5, tol_eigval=1.0e-5):
+has_jax_optax = True
+try:  # pragma: no cover
+    # pylint: disable=unused-import
+    import optax
+    from jax import jit
+    from jax import numpy as jnp
+    from jax import scipy as jsp
+    from jax import value_and_grad
+except (ModuleNotFoundError, ImportError) as e:  # pragma: no cover
+    has_jax_optax = False
+
+# pylint: disable=too-many-arguments, too-many-positional-arguments
+
+
+def factorize(
```

Comment:
This function call several private functions, which is great, but then changes their output internally without being clear why. It will be great if the private functions have the same return type, so the factorize function does not have to modify their returned object.